src/Data/ProgrammesDb/EntityRepository/CoreEntityRepository.php

```diff
@@ -33,6 +33,34 @@ class CoreEntityRepository extends MaterializedPathRepository
 
     private $ancestryCache = [];
 
+    /**
+     * @param int[] $ancestryDbIds
+     * @param string $entityType
+     * @param DateTimeImmutable $fromDateTime
+     * @return int
+     */
+    public function countUpcomingStreamableDescendantsByType(array $ancestryDbIds, string $entityType, DateTimeImmutable $fromDateTime): int
+    {
+        $this->assertEntityType($entityType, ['Clip', 'Episode']);
+
+        $qb = $this->createQueryBuilder('entity')
```

Comment:
You shouldn't be using `$this->createQueryBuilder`, as you're changing the `select()` and `from()` bits. Something like `$this->getEntityManager()->createQueryBuilder()` would be more suitable.

Reply:
That's what it is.... isn't it?